2023/assignments/aryan/aryan.md

```diff
@@ -0,0 +1,204 @@
+
+        It is used to initialize git repository
+
+    **Examples**:
+
+        $ git init
+
+2.  `git add <filename>`
+
+    **Description** :
+
+        It is used to add files or changes in the working directory to the staged area. 
+
+    **Examples**:
+
+        $ git add  hangman.c
+    
+3.  `git status`
+
+    **Description** :
+
+        It is used to display the current status of your git repo by displaying the information about tracked and untracked files in the current working directory. 
+
+    **Examples**:
+
+        $ git status
+
+4.  `git commit -m "descriptive message"`
+
+    **Description** :
+
+        This git commit command captures a snapshot of the project's currently staged changes. 
+
+    **Examples**:
+
+        $ git commit -m "added hangman.c"
+
+5.  `git diff`
+
+    **Description** :
+
+        It shows the changes between the working directory and the last commit. 
+
+    **Examples**:
+
+        $ git diff
+
+6.  `git log`
+
+    **Description** :
+
+        It is used to display the list of commits done in the reverse chronological order meaning the last commit done is shown first. 
+
+    **Examples**:
+
+        $ git log 
+
+7.  `git restore <file-path>`
+
+    **Description** :
+
+        It discards the changes made to a specific file in the working directory and revert it to the stage of last commit. 
+    **Examples**:
+
+        $ git restore project.c
+
+8.  `git remote add <alias> <url>`
+
+    **Description** :
+
+        It is used to add remote repository to our local repository(our working directory).
+
+    **Examples**:
+
+        $ git remote add git@github.com:Silawalaryan/First-Contribution.git
+
+9.  `git push <alias> <branch>`
+
+    **Description** :
+
+        It is used to push certain changes made in the local repository to the added remote repository.
+
+    **Examples**:
+
+        $ git push origin main
+        
+10. `git pull <alias> <branch>`
+
+    **Description** :
+
+        It is used to fetch and download contents from a remote repository and immediately update the local repository to match that content.
+
+    **Examples**:
+
+        $ git pull origin main
+
+11. `git clone <remote-url>`
+
+    **Description** :
+
+        It is used to make a copy of a remote repository in our local repository.
+
+    **Examples**:
+
+        $ git clone git@github.com:Silawalaryan/First-Contribution.git
+
+12. `git branch <branch_name>`
+
+    **Description** :
+
+        It is used to create a branch in git.
+
+    **Examples**:
+
+        $ git branch hanging_person
+
+13. `git branch -m <old_branch_name> <new_branch_name>`
+
+    **Description** :
+
+        It is used to rename a branch in git.
+
+    **Examples**:
+
+        $ git branch -m hanging_person hanging_man
+
+14. `git switch <branch_name>`
+
+    **Description** :
+
+        It is used to switch branches in git.
+
+    **Examples**:
+
+        $ git switch hanging_man
+
+15. `git `push <remote-name> <branch_name>
```

Comment:
backtick is closed ahead.
it shoud have been `git push <remote-branch> <branch_name>` not `git `push <remote-name> <branch_name>